File: Easy/Find the Index of the First Occurrence in a String/find.go
package main

func strStr(haystack string, needle string) int {
    end := 0
    len_needle := len(needle)

    for start := 0; start + len_needle <= len(haystack); start ++{
        end = len_needle + start
        if haystack[start:end] == needle{
            return start
        }
    }
    return -1
}

File: Easy/Find the Index of the First Occurrence in a String/find.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        len_needle = len(needle)
        start = 0
        while start + len_needle <= len(haystack):
            end = start + len_needle
            if haystack[start: end] == needle:
                return start
            start += 1
        return -1
    